src/corner_kick/src/types/theme.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * This file defines the type of the application theme object
+ * @see https://www.styled-components.com/docs/advanced#theming
+ */
+
+/**
+ * The application theme
+ */
+export interface IThemeProvider {
+    colors: IColorProvider;
+}
+
+/**
+ * The color available in the application
+ */
+interface IColorProvider {
+    fg: string;
+    bg: string;
+
+    accent: string;
+    subdued: string;
+    selected: string;
+
+    panel: string;
+    border: string;
+
+    success: string;
+    error: string;
+    warn: string;
+    info: string;
+    debug: string;
+
+    red: string;
```

Comment:
Bit confused about this; seems like `red` should be a value for `fg`, `bg`, etc.? Why are they defined at the same level here?

Reply:
Hmm, some comments here would probably go a long way. If I understand this correctly though, could we change `red`, `orange`, etc. to `red_shade`, `orange_shade`, etc.?

Also, why are they strings? Seems like the shade of red should be defined as a number, or maybe an enum? 

Also, what are `fg` and `bg` then? Not rgb values if `red`, `orange`, etc define the shade?

And what if I set a shade of red and a shade of green?

Sorry for the comment dump above, I think that a few quick comments in the code here would likely address my questions, but just leaving the above brain dump to hopefully help convey where my confusion is coming from.

Reply:
```
// This is the application's color palette. They need
// to be strings as that is what our CSS code accepts.
// Some of those attributes define color for a purpose
// (`fg` defines the color for a foreground elements - like text)
// while others define shades of a particular color (red defines the
// shade of red we use)
```